### PR TITLE
docs: Upgrade doc generating dependencies

### PR DESCRIPTION
**Pull request by d3zd3z**
Addresses CVE-2021-28834
https://github.com/advisories/GHSA-52p9-v744-mwjj

Also removed the explicit jekyll dependency, which according to the
instructions should be commented out if the github-pages dependency is
used.

Signed-off-by: David Brown <david.brown@linaro.org>

### CODE REVIEW

File: docs/Gemfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ source "https://rubygems.org"
 #
 # This will help ensure the proper Jekyll version is running.
 # Happy Jekylling!
-gem "jekyll", ">= 3.7.4"
+# gem "jekyll", ">= 3.7.4"
```
Comment (utzig) [inline docs/Gemfile:11]: Why did you leave it commented out?

**Comment by StefanXhunga** [comment]
Thank you for your assistance!

File: docs/Gemfile (hunk 1 above)
Comment (StefanXhunga) [reply docs/Gemfile:11]: Hello All,
I have two Atlassian 1 accounts in Yahoo and the other in gmail.  The Yahoo account is buggy.  I'm just a self-employed online business person.  I am retired and can not pay any other person.  This is the problem.

Comment (d3zd3z) [reply docs/Gemfile:11]: > Why did you leave it commented out?

I was kind of just leaving the template in place, commenting that line back out, and uncommenting the github pages line. I can remove it.

**Review by utzig — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**